core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -518,15 +529,48 @@ func (c *coordinator) unmarshalLogs(
 func (c *coordinator) ReportWillBeTransmitted(ctx context.Context, report ocr2vrftypes.AbstractReport) error {
 	c.transmittedMu.Lock()
 	defer c.transmittedMu.Unlock()
+
+	blocksRequested := make(map[block]struct{})
+	callbacksRequested := []callback{}
+
+	// Get all requested blocks and callbacks.
 	for _, output := range report.Outputs {
-		c.toBeTransmittedBlocks[block{
+		blocksRequested[block{
 			blockNumber: output.BlockHeight,
 			confDelay:   output.ConfirmationDelay,
 		}] = struct{}{}
 		for _, cb := range output.Callbacks {
-			c.toBeTransmittedCallbacks[cb.RequestID] = struct{}{}
+			callbacksRequested = append(callbacksRequested, callback{
+				blockNumber: cb.BeaconHeight,
+				requestID:   cb.RequestID,
+			})
 		}
 	}
+
+	// Get latest block from log poller.
+	currentHeight, err := c.lp.LatestBlock(pg.WithParentCtx(ctx))
+	if err != nil {
+		return errors.Wrap(err, "Getting latest block in ReportWillBeTransmitted")
+	}
+
+	// Get latest blockhashes from log poller.
+	blockhashesMapping, err := c.getBlockhashesMapping(ctx, uint64(currentHeight-c.lookbackBlocks), uint64(currentHeight))
+	if err != nil {
+		return errors.Wrap(err, "Getting blockhashes in ReportWillBeTransmitted")
+	}
```

Comment:
These queries seem somewhat wasteful if the `blocksRequested` / `callbacksRequested` maps are small (or just the number of unique block numbers that we need to fetch blockhashes for is small), especially if `lookbackBlocks` is large.

Perhaps better to call `GetBlocks` from the log poller with the argument as the exact block numbers you want to get? That'll presumably be a much faster query as well since it'll be paging over less tables in disk.

Reply:
Sure. I was curious about that myself and the sentiment on the call was that local queries such as this would be quick, but I'll cut it down.